src/include/ipc/dai.h

```diff
@@ -74,16 +74,37 @@ struct sof_ipc_dai_config {
 	uint16_t reserved16;	/**< alignment */
 
 	/* reserved for future use */
-	uint32_t reserved[8];
+	uint32_t reserved[4];
+} __attribute__((packed));
+
+struct sof_ipc_multidai_params {
+	struct sof_ipc_cmd_hdr hdr;
+
+	uint32_t num_dais;
+	uint32_t reserved[5];
+
+	struct sof_ipc_stream_map stream_map;
+	struct sof_ipc_dai_base_params params[0];
+} __attribute__((packed));
+
+/* general purpose DAI configuration */
+struct sof_ipc_dai_config {
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_dai_base_params params;
+
+	/* reserved for future use */
+	uint32_t reserved[3];
+	uint32_t num_configs;
```

Comment:
are you positive that this change is compatible with the previous solution? I haven't checked the sizes and it's hard to see with the diff format.

Reply:
Yeah. I ran tests on our side as well.